keel-slack/src/main/kotlin/com/netflix/spinnaker/keel/slack/NotificationEventListener.kt

```diff
@@ -26,21 +32,26 @@ class NotificationEventListener(
   @EventListener(PinnedNotification::class)
   fun onPinnedNotification(notification: PinnedNotification) {
     with(notification) {
-      val envNotifications = repository.environmentNotifications(config.name, pin.targetEnvironment)
-      val deliveryArtifact = pin.reference.let {
-        repository.getArtifact(config.name, it)
-      }
+      val deliveryArtifact = config.artifacts.find {
+        it.reference == pin.reference
+      } ?: return
+
       val pinnedArtifact = repository.getArtifactVersion(deliveryArtifact, pin.version, null)
       val currentArtifact = repository.getArtifactVersionByPromotionStatus(config, pin.targetEnvironment, deliveryArtifact, PromotionStatus.CURRENT)
 
-      envNotifications.forEach { notificationConfig ->
-        if (notificationConfig.type == NotificationType.slack) {
-          val handler = handlers.find {
-            it.type == notification.type
-          } as? SlackNotificationHandler<SlackPinnedNotification>
-          handler?.let {
-            it.sendMessage(SlackPinnedNotification(
-              channel = notificationConfig.address,
+      val handler = handlers.find {
+        it.type == notification.type
+      } as? SlackNotificationHandler<SlackPinnedNotification> ?: return
```

Comment:
Since this is showing up in more places now, might be worth extracting it into that `supporting` function as suggested earlier:
```kotlin
fun <T : SlackNotificationEvent> Collection<SlackNotificationHandler<*>>.supporting(type: NotificationType): SlackNotificationHandler<T> =
  this.find { it. type == type } as? SlackNotificationHandler<T>
```

Reply:
done!